### PR TITLE
feat: Expose `max_row_group_length` option for Parquet writer

### PR DESCRIPTION
**Pull request by disq**

### CODE REVIEW

File: parquet/spec.go [hunk 1]
```diff
@@ -9,13 +9,16 @@ import (
 	"github.com/invopop/jsonschema"
 )
 
+const defaultMaxRowGroupLength = 128 * 1024 * 1024
```
Comment (disq) [inline parquet/spec.go:12]: For some reason this was set to 128 _mibi rows_ but it's actually a row count.

File: parquet/spec.go [hunk 2]
```diff
@@ -94,5 +115,8 @@ func (s *ParquetSpec) Validate() error {
 	if !slices.Contains(allowedRootRepetitions, s.RootRepetition) {
 		return fmt.Errorf("invalid rootRepetition: %s. Allowed values are %s", s.RootRepetition, strings.Join(allowedRootRepetitions, ", "))
 	}
+	if s.MaxRowGroupLength != nil && *s.MaxRowGroupLength <= 0 {
+		return fmt.Errorf("invalid: maxRowGroupLength: %v. Must be zero or positive", *s.MaxRowGroupLength)
```
Comment (marianogappa) [inline parquet/spec.go:119]: Is zero or a small number ok?

**Review by marianogappa — APPROVED** [review]

File: parquet/spec.go (hunk 2 above)
Comment (disq) [reply parquet/spec.go:119]: Zero would be going back to the default of the arrow parquetwriter. Updated this check to be `< 0`

**Review by erezrokah — COMMENTED** [review]
FYI changing this setting works see results with 1000 as the value on a table with more than 1000 rows 
image

**Pull request merged**